frontend/src/scenes/billing/billingLogic.ts

```diff
@@ -319,6 +257,82 @@ export const billingLogic = kea<billingLogicType>([
                 router.actions.replace('/organization/billing')
             }
             actions.registerInstrumentationProps()
+
+            if (values.productSpecificAlert) {
```

Comment:
This is a lot of code to shove into the listener, I think it would make more sense to have this as a separate action/listener. So when billing loads we know we `registerInstrumentationProps` and `setBillingAlert` without having to wonder what this big block of code is all about. In fact this could just to in the setBillingAlert reducer - when that is called, we run this block and return the appropriate value.

Reply:
Agree, nicer to separate out everything here

Reply:
got it, thanks.

Reply:
moved into a separate listener since we can't dispatch actions from within reducers.